<div class="content"><p> </p><p>The <b>R_DhcpV4FailoverDeleteRelationship</b> method is used
to delete an existing failover relationship on the DHCPv4 server.</p><dl>
<dd>
<div><pre> DWORD R_DhcpV4FailoverDeleteRelationship(
   [in, unique, string] DHCP_SRV_HANDLE ServerIpAddress,
   [in, string, unique] LPWSTR pRelationshipName
 );
</pre></div>
</dd></dl><p><b>ServerIpAddress: </b>The IP address/host name of
the DHCP server. This parameter is unused.</p><p><b>pRelationshipName: </b>A pointer to a
null-terminated Unicode string that contains the name of the failover
relationship to be deleted.</p><p><b>Return Values: </b>A 32-bit unsigned integer value
that indicates return status. The return value ERROR_SUCCESS (0x00000000)
indicates that the operation was completed successfully. Otherwise, it contains
a Win32 error code, as specified in <span><a href="../ms-erref/1bc92ddf-b79e-413c-bbaa-99a5281a6c90" data-linktype="relative-path">[MS-ERREF]</a></span>.
This error code value can correspond to a DHCP-specific failure, which takes a
value between 20000 and 20123, or any generic failure.</p><dl>
<dd>
<table><thead>
  <tr>
   <th>
   <p>Return value/code</p>
   </th>
   <th>
   <p>Description</p>
   </th>
  </tr>
 </thead><tbody><tr>
  <td>
  <p>0x00000000</p>
  <p>ERROR_SUCCESS</p>
  </td>
  <td>
  <p>The call was successful.</p>
  </td>
 </tr><tr>
  <td>
  <p>0x00004E92</p>
  <p>ERROR_DHCP_FO_RELATIONSHIP_DOES_NOT_EXIST</p>
  </td>
  <td>
  <p>The failover relationship doesn&#39;t exist.</p>
  </td>
 </tr></tbody></table>
</dd></dl><p>The opnum field value for this method is 91.</p><p>When processing this call, the DHCP server MUST do the
following:</p><ul><li><p><span><span> 
</span></span>If <i>pRelationshipName</i> parameter is NULL, return
ERROR_INVALID_PARAMETER.</p>
</li><li><p><span><span> 
</span></span>Validate whether this method is authorized for read/write access
as specified in section <span><a href="38840bd0-0a5a-47ed-bc69-0dd870d33aec" data-linktype="relative-path">3.5.5</a></span>. If not,
return ERROR_ACCESS_DENIED.</p>
</li><li><p><span><span> 
</span></span>Iterate through the elements in <b>DHCPv4FailoverRelationshipList</b>
ADM element and retrieve the <b>DHCPv4FailoverRelationship</b> ADM element
corresponding to the <i>pRelationshipName</i> parameter. If the corresponding <b>DHCPv4FailoverRelationship</b>
ADM element is not found, return ERROR_DHCP_FO_RELATIONSHIP_DOES_NOT_EXIST.</p>
</li><li><p><span><span> 
</span></span>For each of the IPv4 subnet address configured as part of
retrieved <b>DHCPv4FailoverRelationship</b> ADM element, retrieve the
corresponding <b>DHCPv4Scope</b> ADM element and set <b>DHCPv4Scope.IsFailover</b>
to FALSE.</p>
</li><li><p><span><span> 
</span></span>Delete the retrieved <b>DHCPv4FailoverRelationship</b> ADM
element.</p>
</li><li><p><span><span> 
</span></span>Return ERROR_SUCCESS.</p>
</li></ul><p><b>Exceptions Thrown:</b> No exceptions are thrown beyond
those thrown by the underlying RPC protocol <span><a href="../ms-rpce/290c38b1-92fe-4229-91e6-4fc376610c15" data-linktype="relative-path">[MS-RPCE]</a></span>.</p></div>